src/Model/ModelTrait.php

```diff
@@ -151,10 +150,11 @@ private function getFirstParameter(string $method): \ReflectionParameter
      */
     private function constructClasses(\ReflectionParameter $param, mixed $value, bool $isCollection = false): mixed
     {
-        $class = $param->getType() && !$param->getType()->isBuiltin()
-            ? new \ReflectionClass($param->getType()->getName()) : null;
+        $classType = $param->getType();
+        $class = $classType && is_a($classType, "ReflectionNamedType") && !$classType->isBuiltin()
+            ? new \ReflectionClass($classType->getName()) : null;
 
-        if (!empty($class)) {
+        if (is_a($class, "ReflectionClass")) {
```

Comment:
I'd use `instanceof` instead of `is_a()`

Reply:
Fixed